src/coreclr/System.Private.CoreLib/src/System/Reflection/MdFieldInfo.cs

```diff
@@ -33,8 +33,7 @@ internal override bool CacheEquals(object? o)
             return
                 o is MdFieldInfo m &&
                 m.m_tkField == m_tkField &&
-                m_declaringType.TypeHandle.GetModuleHandle().Equals(
-                    m.m_declaringType.TypeHandle.GetModuleHandle());
+                ReferenceEquals(m_declaringType, m.m_declaringType);
```

Comment:
Does the `MdFieldInfo` needs overridden GetHashCode and Equals? It is part of the same caches as RuntimeFieldInfo, so I think it should.

Reply:
`RuntimeFieldInfo` is the parent of `MdFieldInfo` and `RtFieldInfo` is that necessary?

Reply:
I have missed this detail. Ignore this comment.

Reply:
`RtFieldInfo.CacheEquals` compares handles, so probably same as MethodInfo not needed
       internal override bool CacheEquals(object? o)
        {
            return o is RtFieldInfo m && m.m_fieldHandle == m_fieldHandle;
        }